### PR TITLE
CI: Enable Python 3.10 in full CPU testing

### LINKED ISSUE #9980: Support Python 3.10
## 🚀 Feature
We should support Python 3.10 as Python 3.10 was released at the beginning of this month.

**Required dependencies supporting Python 3.10**
- [x] numpy>=1.21.0 https://numpy.org/news/#numpy-1210-release
- [x] torch>=1.11.0 https://github.com/pytorch/pytorch/issues/66424
- [x] tqdm
- [x] PyYAML
- [x] fsspec
- [x] tensorboard https://github.com/tensorflow/tensorboard/releases/tag/2.8.0
- [x] torchmetrics since its dependency `torch>=1.11` supports Python 3.10
- [x] pyDeprecate https://github.com/Borda/pyDeprecate/issues/18
- [x] packaging
- [x] typing-extensions

The above checklist is mainly based on PyPI badge on pypi.org
______________________________________________________________________

#### If you enjoy Lightning, check out our other projects! ⚡

<sub>

- [**Metrics**](https://github.com/PyTorchLightning/metrics): Machine learning metrics for distributed, scalable PyTorch applications.

- [**Flash**](https://github.com/PyTorchLightning/lightning-flash): The fastest way to get a Lightning baseline! A collection of tasks for fast prototyping, baselining, finetuning and solving problems with deep learning

- [**Bolts**](https://github.com/PyTorchLightning/lightning-bolts): Pretrained SOTA Deep Learning models, callbacks and more for research and production with PyTorch Lightning and PyTorch

- [**Lightning Transformers**](https://github.com/PyTorchLightning/lightning-transformers): Flexible interface for high performance research using SOTA Transformers leveraging Pytorch Lightning, Transformers, and Hydra.

</sub>

cc @borda @carmocca @akihironitta @tchaton @rohitgr7

### PR DESCRIPTION
**Pull request by akihironitta**
## What does this PR do?
Part of #9980.
Adds Python 3.10 only for "full" cpu testing.

### Does your PR introduce any breaking changes? If yes, please list them.
None

## Before submitting

- [x] Was this **discussed/approved** via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [x] Did you make sure to **update the documentation** with your changes? (if necessary)
- [n/a] Did you write any **new necessary tests**? (not for typos and docs)
- [n/a] Did you verify new and **existing tests pass** locally with your changes?
- [n/a] Did you list all the **breaking changes** introduced by this pull request?
- [n/a] Did you **update the CHANGELOG**? (not for typos, docs, test updates, or minor internal changes/refactors)

## PR review

Anyone in the community is welcome to review the PR.
Before you start reviewing, make sure you have read the [review guidelines](https://github.com/Lightning-AI/lightning/wiki/Review-guidelines). In short, see the following bullet-list:

- [x] Is this pull request ready for review? (if not, please submit in draft mode)
- [x] Check that all items from **Before submitting** are resolved
- [x] Make sure the title is self-explanatory and the description concisely explains the PR
- [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?

Make sure you had fun coding 🙃

cc @carmocca @akihironitta @borda

### CODE REVIEW

**Comment by akihironitta** [comment]
Still wip!

**Review by rohitgr7 — APPROVED** [review]

**Review by carmocca — APPROVED** [review]

**Review by awaelchli — APPROVED** [review]
Good news!

**Review by Borda — APPROVED** [review]

**Review by otaj — APPROVED** [review]
